packages/docmanager-extension/src/index.tsx

```diff
@@ -117,16 +117,17 @@ const openerPlugin: JupyterFrontEndPlugin<IDocumentWidgetOpener> = {
           shell.add(widget, 'main', options || {});
         }
         shell.activateById(widget.id);
-        this.opened.emit(widget);
-      },
+        this._opened.emit(widget);
+      }
+
       get opened() {
-        if (this._signal) {
-          return this._signal;
-        }
-        this._signal = new Signal<IDocumentWidgetOpener, IDocumentWidget>(this);
-        return this._signal;
+        return this._opened;
       }
-    });
+
+      private _opened = new Signal<IDocumentWidgetOpener, IDocumentWidget>(
+        opener
```

Comment:
It should be this, isn't it?

```suggestion
        this
```

Reply:
yes fixed in https://github.com/jupyterlab/jupyterlab/pull/12701/commits/098660f8e321662c88994436e59849f161f750ce